packages/client-nestjs/vitest.config.ts

```diff
@@ -22,6 +22,15 @@ export default defineConfig({
           include: ["src/**/*.integration.spec.ts"],
         },
       },
+      {
+        test: {
+          name: "typecheck",
+          include: ["src/**/*.test-d.ts"],
+          typecheck: {
+            enabled: true,
+          },
+        },
+      },
```

Comment:
maybe it is not a good idea to introduce a new project for type testing, maybe we should include those tests into unit tests

but we can keep different extensions